docs/source/guides/model-loading-and-management.rst

```diff
@@ -104,9 +104,11 @@ BentoML provides an efficient mechanism for loading AI models to accelerate mode
              def __init__(self):
                  self.iris_model = joblib.load(self.iris_ref.path_of("model.pkl"))
 
-.. important::
+When using ``HuggingFaceModel`` and ``BentoModel``, you must load the model from the class scope of a Service. Defining the model as a class variable declares it as a dependency of the Service, ensuring the models are referenced by the Bento when transported and deployed. If you call these two APIs within the constructor of a Service class, the model will not be referenced by the Bento. As a result, it will not be pushed or deployed, leading to a model ``NotFound`` error.
```

Comment:
we should probably explain the difference when `BentoModel` and `HuggingFaceModel` is defined in class scope.

By default, `__get__` from `BentoModel` returns the `bentoml.Model` object, whereas `HuggingFaceModel` yields a string, so users can do this directly

```python
@bentoml.service()
class Service:
  llama_path = HuggingFaceModel("meta-llama/Llama-3.2-3B-Instruct")
  
  def __init__(self):
    self.model = transformers.AutoModelForCausalLM.from_pretrained(self.llama_path)
```